compiler/rustc_middle/src/lint.rs

```diff
@@ -272,11 +272,13 @@ pub fn struct_lint_level<'s, 'd>(
             // emit shouldn't be automatically fixed by rustfix.
             err.allow_suggestions(false);
 
-            // If this is a future incompatible lint it'll become a hard error, so
-            // we have to emit *something*. Also, if this lint occurs in the
-            // expansion of a macro from an external crate, allow individual lints
-            // to opt-out from being reported.
-            if future_incompatible.is_none() && !lint.report_in_external_macro {
+            // If this is a future incompatible that is not an edition fixing lint
+            // it'll become a hard error, so we have to emit *something*. Also,
+            // if this lint occurs in the expansion of a macro from an external crate,
+            // allow individual lints to opt-out from being reported.
+            let not_future_incompatible =
```

Comment:
The edition fixing does not seem obviously relevant here. Can you say more about this change being necessary (and why it's part of this PR, vs. something separate - it feels likely orthogonal?)

Reply:
This PR changes some lints to have "future incompatible" info that is tied to an edition. That causing CI failures and other problems where those lints were now being emitted even from foreign macros.

Reply:
Ah, I see. Makes sense. I think part of my concern arose from the variable name being a bit awkward (negatives are always annoying), but I don't have good suggestions. Seems all good.